src/models/tiles/text/text-content.ts

```diff
@@ -26,25 +26,33 @@ export const TextContentModel = TileContentModel
     editor:  undefined as CustomEditor | undefined,
   }))
   .views(self => ({
-    get joinText() {
+    // guarantees string (not readonly string) types
+    get textStr(): string | string[] {
       return Array.isArray(self.text)
-        ? self.text.join("\n")
-        : self.text as string;
+              ? self.text as string[]
+              : self.text as string;
```

Comment:
This is pretty annoying. I suspect this really should be a bug in the MST type system, but oh well.

Reply:
I agree. It was [this PR](https://github.com/mobxjs/mobx-state-tree/pull/1936) which caused the trouble. I wondered if perhaps this
```typescript
export class ArrayType<IT extends IAnyType> extends ComplexType<
    readonly IT["CreationType"][] | undefined,
```
should really be this
```typescript
export class ArrayType<IT extends IAnyType> extends ComplexType<
    readonly IT["CreationType"][] | IT["CreationType"][] | undefined,
```
but I didn't take the time to pursue it.